sms_sponsorship/models/recurring_contract.py

```diff
@@ -0,0 +1,103 @@
+from odoo import _, api, models
+
+
+class RecurringContract(models.Model):
+    _inherit = 'recurring.contract'
+
+    @api.model
+    def _get_sponsorship_standard_lines(self):
+        """ Select Sponsorship and General Fund by default """
+        res = []
+        product_obj = self.env['product.product'].with_context(lang='en_US')
+        sponsorship_id = product_obj.search(
+            [('name', '=', 'Sponsorship')])[0].id
+        gen_id = product_obj.search(
+            [('name', '=', 'General Fund')])[0].id
+        sponsorship_vals = {
+            'product_id': sponsorship_id,
+            'quantity': 1,
+            'amount': 42,
+            'subtotal': 42
+        }
+        gen_vals = {
+            'product_id': gen_id,
+            'quantity': 1,
+            'amount': 8,
+            'subtotal': 8
+        }
+        res.append([0, 6, sponsorship_vals])
+        res.append([0, 6, gen_vals])
+        return res
+
+    @api.model
+    def create_sms_sponsorship(self, vals, utm_source, utm_medium, utm_campaign, partner, sms_child_request):
+        if not partner:
+            # Search for existing partner
+            partner = self.env['res.partner'].search([
+                ('firstname', 'ilike', vals['firstname']),
+                ('lastname', 'ilike', vals['lastname']),
+                ('email', '=', vals['email'])
+            ])
+            sms_child_request.partner_id = partner
+
+        if not partner:
+            partner = self.env['res.partner'].create({
+                'firstname': vals['firstname'],
+                'lastname': vals['lastname'],
+                'phone': vals['phone'],
+                'email': vals['email'],
+            })
+            sms_child_request.partner_id = partner
+
+        # Check origin
+        internet_id = self.env.ref('utm.utm_medium_website').id
+        utms = self.env['utm.mixin'].get_utms(
```

Comment:
get_utms is not accessible from this module. Instead you should directly receive the `id` of the related objects as we don't let the user choose it. I will discuss this with you.